### ISSUE TITLE
Couldn't extract salt - System.IO.FileNotFoundException

### ISSUE THREAD

**Issue by RexTheCapt** [rfvgyhn/min-ed-launcher#55]
Every single time i try to run min-ed, i keep getting the same error. Got the same error before updating to v0.6.0. Is there a fix for this?

I am running windows 11.

Log:
```
[18:43:55 INF] Elite Dangerous: Minimal Launcher - v0.6.0+8103a3a6
[18:43:56 ERR] Couldn't extract salt - System.IO.FileNotFoundException:
IO_FileName_Name, C:\Users\user\ClientSupport.dll
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr , String , String , ObjectHandleOnStack )
   at System.Runtime.Loader.AssemblyLoadContext.InternalLoadFromPath(String , String )
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String )
   at System.Reflection.Assembly.LoadFrom(String )
   at MinEdLauncher.Cobra.getSalt@42.Invoke(Unit unitVar)
```

**Comment by rfvgyhn** [comment]
There might be a bug when trying to resolve the path to ClientSupport.dll. I won't have access to a computer for a few days to check though. A couple things you could try until then:

* Ensure MinEdLauncher.exe is in the same folder as EdLaunch.exe.
* Try setting the `gameLocation` property in your settings file to the path that EdLaunch.exe is in.
* Posting the full log could help me narrow down the issue.

**Comment by RexTheCapt** [comment]
I did as your instructions said.

My console output is:
```
[12:10:32 INF] Elite Dangerous: Minimal Launcher - v0.6.0+8103a3a6
[12:10:33 ERR] Couldn't extract salt - System.IO.FileNotFoundException:
IO_FileName_Name, C:\Users\user\ClientSupport.dll
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr , String , String , ObjectHandleOnStack )
   at System.Runtime.Loader.AssemblyLoadContext.InternalLoadFromPath(String , String )
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String )
   at System.Reflection.Assembly.LoadFrom(String )
   at MinEdLauncher.Cobra.getSalt@42.Invoke(Unit unitVar)

[process exited with code 1 (0x00000001)]
```

Here is my log file (censored my username in filepaths):
[min-ed-launcher.censored.log](https://github.com/rfvgyhn/min-ed-launcher/files/9591433/min-ed-launcher.censored.log)

**Comment by rfvgyhn** [comment]
So it looks like you have the normal launcher installed at C:\Users\[user], is that correct? Does the `ClientSupport.dll` file exist there too?

**Comment by RexTheCapt** [comment]
The launcher is downloaded thru steam, but I usually just start it standalone without steam most of the time due to a script I have. I do not have any DLL files in my root user directory.

**Comment by rfvgyhn** [comment]
What triggers your script to run, what's the path to it and how does your script start the launcher? This looks like a working directory issue.

**Comment by RexTheCapt** [comment]
It’s a manually ran script whenever I want to start the game, I have multiple accounts that I run. The command looks like this:

```PS

$Host.UI.RawUI.WindowTitle = "ED <accountName>"

Start-Process -FilePath '<SteamLibrary>\\steamapps\\common\\Elite Dangerous\\MinEdLauncher.exe' -ArgumentList "/frontier <fdevAccount> /autoquit"

```

I have placed it in a directory I use for all my scripts, it is also added top the system path variable so I can run it anywhere. I have just noticed while writing this that the default location in powershell (or WindowsTerminal) is C:\Users\<name>

 

I added -WorkingDirectory <pathToEd> to Start-Process and that fixed the issue… thank you for pointing that out.
